src/components/application_manager/test/hmi_capabilities_test.cc

```diff
@@ -286,84 +283,78 @@ class HMICapabilitiesTest : public ::testing::Test {
+        .WillByDefault(ReturnRef(mock_application_manager_settings_));
     ON_CALL(mock_application_manager_settings_, hmi_capabilities_file_name())
-        .WillByDefault(ReturnRef(file_name_));
+        .WillByDefault(ReturnRef(kHmiCapabilitiesDefaultFile));
     ON_CALL(mock_application_manager_settings_,
             hmi_capabilities_cache_file_name())
-        .WillByDefault(ReturnRef(file_cache_name_));
-    EXPECT_CALL(mock_event_dispatcher, add_observer(_, _, _));
-    EXPECT_CALL(mock_event_dispatcher, remove_observer(_));
-    ON_CALL(mock_application_manager_settings_, launch_hmi())
-        .WillByDefault(Return(false));
-    hmi_capabilities_test =
+        .WillByDefault(ReturnRef(kHmiCapabilitiesCacheFile));
+
+    hmi_capabilities_ =
         std::make_shared<HMICapabilitiesForTesting>(mock_app_mngr_);
-    hmi_capabilities_test->Init(last_state_wrapper_);
   }
 
   void TearDown() OVERRIDE {
-    DeleteFileIfExists(kHmiCapabilitiesCacheDefaultFile);
-    hmi_capabilities_test.reset();
-  }
-  static void TearDownTestCase() {
+    DeleteFileIfExists(kHmiCapabilitiesCacheFile);
     DeleteFileIfExists(kAppInfoDataFile);
+    hmi_capabilities_.reset();
   }
 
-  static void DeleteFileIfExists(const std::string& file_path) {
+  void CreateFile(const std::string& file_name) {
+    file_system::CreateFile(file_name);
+    ASSERT_TRUE(file_system::FileExists(file_name));
+  }
+
+  void DeleteFileIfExists(const std::string& file_path) {
     if (file_system::FileExists(file_path)) {
       EXPECT_TRUE(::file_system::DeleteFile(file_path));
     }
   }
 
-  void SetCooperating();
+  std::shared_ptr<HMICapabilitiesForTesting> hmi_capabilities_;
   NiceMock<MockApplicationManager> mock_app_mngr_;
-  NiceMock<event_engine_test::MockEventDispatcher> mock_event_dispatcher;
-  resumption::LastStateWrapperPtr last_state_wrapper_;
+  NiceMock<event_engine_test::MockEventDispatcher> mock_event_dispatcher_;
   NiceMock<MockApplicationManagerSettings> mock_application_manager_settings_;
-  std::shared_ptr<HMICapabilitiesForTesting> hmi_capabilities_test;
-  const std::string file_name_;
-  const std::string file_cache_name_;
-  application_manager_test::MockRPCService mock_rpc_service_;
+  resumption::LastStateWrapperPtr last_state_wrapper_;
+  smart_objects::CSmartSchema schema_;
 };
 
 TEST_F(HMICapabilitiesTest, LoadCapabilitiesFromFile) {
-  DeleteFileIfExists(kAppInfoDataFile);
+  ON_CALL(mock_app_mngr_, IsSOStructValid(_, _)).WillByDefault(Return(true));
 
   EXPECT_CALL(*(MockMessageHelper::message_helper_mock()),
               CommonLanguageFromString(_))
       .WillRepeatedly(Invoke(TestCommonLanguageFromString));
   EXPECT_CALL(*(MockMessageHelper::message_helper_mock()),
               CommonLightNameFromString(_))
       .WillRepeatedly(Invoke(TestCommonLightNameFromString));
-  ASSERT_TRUE(hmi_capabilities_test->LoadCapabilitiesFromFile());
+
+  hmi_capabilities_->Init(last_state_wrapper_);
```

Comment:
@ydementieiev I would rename it to `hmi_capabilities_for_testing` as it is different from real HMI capabilities.

Reply:
@AByzhynar  Now we using real HMICapabilities class.